allennlp/training/optimizers.py

```diff
@@ -75,6 +80,10 @@ def make_parameter_groups(
     The dictionary's return type is labeled as `Any`, because it can be a `List[torch.nn.Parameter]`
     (for the "params" key), or anything else (typically a float) for the other keys.
     """
+    # The RegexOptimizer passes in the `groups` argument as False so this function is a no-op.
+    if groups is False:
```

Comment:
I'm not super happy about altering the type of `groups` (e.g. can now be `Any`) but it seemed to be the least invasive way of not touching any of the code when the individual optimizers are initialized (where they call this function with the `model_parameters` and `parameter_groups`)

Reply:
Glad that you found a better way to do this!